processor/tailsamplingprocessor/internal/sampling/span_count_sampler.go

```diff
@@ -22,24 +22,33 @@ import (
 type spanCount struct {
 	logger   *zap.Logger
 	minSpans int32
+	maxSpans int32
 }
 
 var _ PolicyEvaluator = (*spanCount)(nil)
 
 // NewSpanCount creates a policy evaluator sampling traces with more than one span per trace
-func NewSpanCount(logger *zap.Logger, minSpans int32) PolicyEvaluator {
+func NewSpanCount(logger *zap.Logger, minSpans, maxSpans int32) PolicyEvaluator {
 	return &spanCount{
 		logger:   logger,
 		minSpans: minSpans,
+		maxSpans: maxSpans,
 	}
 }
 
 // Evaluate looks at the trace data and returns a corresponding SamplingDecision.
 func (c *spanCount) Evaluate(_ pcommon.TraceID, traceData *TraceData) (Decision, error) {
 	c.logger.Debug("Evaluating spans counts in filter")
 
-	if int(traceData.SpanCount.Load()) >= int(c.minSpans) {
+	spanCount := traceData.SpanCount.Load()
+	switch {
+	case c.maxSpans == 0 && int(spanCount) >= int(c.minSpans):
 		return Sampled, nil
+	case c.minSpans == 0 && int(spanCount) <= int(c.maxSpans):
+		return Sampled, nil
+	case int(spanCount) >= int(c.minSpans) && int(spanCount) <= int(c.maxSpans):
```

Comment:
Do we need the other cases?  Does this case catch everything within the range if we use `> min` and `<= max`?

Reply:
Other cases are for the user to only set one of `min_spans` and `max_spans` config options. Assuming that only `min_spans` is set, `max_spans` will be empty value 0. In this case, it is wrong to use `>=min` and `<=max`